### PR TITLE
Synthetic diffusion MGN example.

### PR DESCRIPTION
**Pull request by YohannDudouit**
**Summary**
This PR adds a pure-Torch MGN-like graph diffusion validation workflow for AMS homogeneous graph surrogates. It trains a small synthetic diffusion model, exports an AMS-compatible TorchScript model, writes runtime fixtures, and validates C++ AMS inference against Python TorchScript reference output.

**What Changed**
- Added `generate_mgn_graph_diffusion.py` with separate modes:
  - `feasibility`: eager/script/reload checks on dynamic graph sizes
  - `train`: trains the synthetic graph diffusion surrogate and saves checkpoint/metrics
  - `fixtures`: exports TorchScript and writes runtime fixture binaries plus `fixtures.json`
- Added `test_graph_mgn_surrogate.cpp` to load runtime fixtures and compare AMS `node:delta_u` output against Python TorchScript references for `N=24` and `N=73`.
- Added `MGN_DIFFUSION_*` CTest workflow entries with dependency ordering:
  - feasibility → train → fixtures → AMS parity

**Testing**
```bash
python3 -m py_compile tests/AMSlib/models/generate_mgn_graph_diffusion.py
git diff --check
ctest --test-dir build -R MGN_DIFFUSION -j 8 --output-on-failure
```

Normal CTest can skip this training workflow with:

```bash
ctest --test-dir build -LE MGN_DIFFUSION --output-on-failure
```

Note: This PR is built on top of #194 and therefore include the graph construction and unit tests, until #194 is merged.

### CODE REVIEW

File: tests/AMSlib/ams_interface/test_graph_fallback.cpp [hunk 1]
```diff
@@ -124,21 +124,9 @@ CATCH_TEST_CASE("AMSHomogeneousGraph validates construction",
   AMSInit();
 
   CATCH_REQUIRE_NOTHROW(makeValidGraph());
-  {
```
Comment (lpottier) [inline tests/AMSlib/ams_interface/test_graph_fallback.cpp:127]: @YohannDudouit Why were these tests/assertions deleted Is that a bad rebase?

File: tests/AMSlib/ams_interface/CMakeLists.txt [hunk 2]
```diff
@@ -43,3 +43,21 @@ ADD_AMS_UNIT_TEST(AMS_GRAPH_FALLBACK ams_graph_fallback)
...
+)
+
+# Normal parity testing consumes the checked-in TorchScript model and
+# self-contained JSON fixtures; training/export is an independent opt-in path.
+add_test(
+  NAME MGN_DIFFUSION_AMS_PARITY
+  COMMAND $<TARGET_FILE:ams_graph_mgn_surrogate> -s --reporter console
+)
+set_tests_properties(MGN_DIFFUSION_AMS_PARITY PROPERTIES
+  LABELS "AMS_INTERFACE;MGN_DIFFUSION"
```
Comment (lpottier) [inline tests/AMSlib/ams_interface/CMakeLists.txt:61]: So as `MGN_DIFFUSION_AMS_PARITY` is flagged as `MGN_DIFFUSION` when we want to skip the training step with `ctest -LE MGN_DIFFUSION` then it also skips the parity test. Maybe we have another category like `MGN_DIFFUSION_TRAINING` for the training only and we leave the parity test to `AMS_INTERFACE`?

File: tests/AMSlib/models/CMakeLists.txt [hunk 3]
```diff
@@ -114,3 +115,122 @@ configure_file(
   ${CMAKE_CURRENT_BINARY_DIR}/ams_test_linear_models.hpp
   COPYONLY
 )
+
+find_program(AMS_PYTHON_EXECUTABLE NAMES python3 python REQUIRED)
```
Comment (lpottier) [inline tests/AMSlib/models/CMakeLists.txt]: That should not be needed as we already find Python in the tests/CMake. So you should have access to `Python3_EXECUTABLE` which is named `AMS_PYTHON_EXECUTABLE` here

File: CMakeLists.txt [hunk 4]
```diff
@@ -40,6 +40,7 @@ option(ENABLE_CUDA "Enable CUDA support" OFF)
 option(ENABLE_HIP "Enable HIP support" OFF)
 option(ENABLE_CALIPER "Use Caliper for profiling" OFF)
 option(ENABLE_TESTS "Build AMS tests" OFF)
+option(ENABLE_TRAINING_TESTS "Enable long-running model training tests" OFF)
```
Comment (lpottier) [inline CMakeLists.txt:43]: Minor but `ENABLE_TRAINING_TESTS=On` with `ENABLE_TESTS=Off` silently does nothing, we should add a `message(WARNING ...)`

**Review by lpottier — CHANGES_REQUESTED** [review]
LGTM overall, I've added a few comments. The only thing that is annoying in the current version is the dependency to `torch`. `MGN_DIFFUSION_FEASIBILITY` runs by default, so it makes a Python torch install a requirement at test time. Can we deactivate it by default or maybe check for `torch` like we do for `hdf5` in `tests/CMakeLists.txt` and if torch is not there we mark these tests `DISABLED`?

File: tests/AMSlib/ams_interface/test_graph_fallback.cpp (hunk 1 above)
Comment (YohannDudouit) [reply tests/AMSlib/ams_interface/test_graph_fallback.cpp:127]: I indeed think that something weird happened in a rebase or something similar, reverted in [b57b447](https://github.com/llnl/AMS/pull/198/commits/b57b4479ca56e32b5e6ab523312db3928a4174db)

File: tests/AMSlib/ams_interface/CMakeLists.txt (hunk 2 above)
Comment (YohannDudouit) [reply tests/AMSlib/ams_interface/CMakeLists.txt:61]: Done in [c2d3d3d](https://github.com/llnl/AMS/pull/198/commits/c2d3d3d4a9bc787cedeead3173569820af6bfdef). I left the label MGN_DIFFUSION to all these tests and added the label MGN_DIFFUSION_TRAINING to MGN_DIFFUSION_FIXTURES and MGN_DIFFUSION_TRAIN.

File: tests/AMSlib/models/CMakeLists.txt (hunk 3 above)
Comment (YohannDudouit) [reply tests/AMSlib/models/CMakeLists.txt]: Done in [560f35c](https://github.com/llnl/AMS/pull/198/commits/560f35c852b45dc18f56627b6f162086cd135dd9).

File: CMakeLists.txt (hunk 4 above)
Comment (YohannDudouit) [reply CMakeLists.txt:43]: Added the warning in [e0c99a6](https://github.com/llnl/AMS/pull/198/commits/e0c99a601c4b02a1aa28b12800cd70b629e471e7).

**Review by lpottier — APPROVED** [review]
LGTM
